pkg/log/log.go

```diff
@@ -0,0 +1,201 @@
+    case "warn", "warning":
+        return log.WarnLevel
+    case "debug":
+        return log.DebugLevel
+    case "info":
+        return log.InfoLevel
+    }
+    return defaultLogLevel
+}
+
+// hook injects file name and line pos into log entry.
+type hook struct{}
+
+// Fire implements logrus.Hook interface
+func (h *hook) Fire(entry *log.Entry) error {
+    // these two num are set by manually testing
+    pc := make([]uintptr, 4)
+    cnt := runtime.Callers(10, pc)
+
+    for i := 0; i < cnt; i++ {
+        fu := runtime.FuncForPC(pc[i] - 1)
+        name := fu.Name()
+        if !isSkippedPackageName(name) {
+            file, line := fu.FileLine(pc[i] - 1)
+            entry.Data["file"] = path.Base(file)
+            entry.Data["line"] = line
+            break
+        }
+    }
+    return nil
+}
+
+// Levels implements logrus.Hook interface.
+func (h *hook) Levels() []log.Level {
+    return log.AllLevels
+}
+
+// isSKippedPackageName tests wether path name is on log library calling stack.
+func isSkippedPackageName(name string) bool {
+    return strings.Contains(name, "github.com/sirupsen/logrus") ||
+            strings.Contains(name, "github.com/apache/pulsar-client-go/pkg/log")
+}
+
+// formatter is for compatibility with ngaut/log
+type formatter struct {
+    DisableTimestamp, EnableEntryOrder bool
+}
+
+// Format implements logrus.Formatter
+func (f *formatter) Format(entry *log.Entry) ([]byte, error) {
+    var b *bytes.Buffer
+    if entry.Buffer != nil {
+        b = entry.Buffer
+    } else {
+        b = &bytes.Buffer{}
+    }
+
+    if !f.DisableTimestamp {
+        fmt.Fprintf(b, "%s ", entry.Time.Format(defaultLogTimeFormat))
+    }
+    if file, ok := entry.Data["file"]; ok {
+        fmt.Fprintf(b, "%s:%v:", file, entry.Data["line"])
+    }
+    fmt.Fprintf(b, " [%s] %s", entry.Level.String(), entry.Message)
+
+    if f.EnableEntryOrder {
+        keys := make([]string, 0, len(entry.Data))
+        for k := range entry.Data {
+            if k != "file" && k != "line" {
+                keys = append(keys, k)
+            }
+        }
+        sort.Strings(keys)
+        for _, k := range keys {
+            fmt.Fprintf(b, " %v=%v", k, entry.Data[k])
+        }
+    } else {
+        for k, v := range entry.Data {
+            if k != "file" && k != "line" {
+                fmt.Fprintf(b, " %v=%v", k, v)
+            }
+        }
+    }
+
+    b.WriteByte('\n')
+
+    return b.Bytes(), nil
+}
+
+// SetLevel sets log's level by a level string.
+func SetLevel(level string) {
+    log.SetLevel(stringToLogLevel(level))
+}
+
+// GetLevel gets current log's level as a level string.
+func GetLevel() string {
+    return log.GetLevel().String()
+}
+
+// SetOutput sets the filename for the log.
+func SetOutput(filename string) {
+    output := &lumberjack.Logger{
+        Filename:  filename,
+        LocalTime: true,
+    }
+    log.SetOutput(output)
+}
+
+func init() {
+    log.SetLevel(defaultLogLevel)
```

Comment:
This is a library so we shouldn't set any specific logger formatting property.

We should just leave the application free to configure through Logrus and point then to that documentation.